buffer/test/com/treode/buffer/PagedBufferSpec.scala

```diff
@@ -48,33 +48,52 @@ class PagedBufferSpec extends FlatSpec {
   capacity (64, 2)
   capacity (65, 3)
 
-  def discard (nbytes: Int, npages: Int) {
-    it should (s"discard the right pages for nbytes=$nbytes") in {
+  def discard (nbytes: Int, npages: Int, wbytes: Int, rbytes: Int, dbytes: Int) {
```

Comment:
I am trying to test if the return value is proper with different values of wbytes (initial write position), rbytes (initial read position) and removing dbytes bytes after initializing the two. Taking all 3 values as arguements gives more control, since in the function, initial writepos is always set to (4<<pageBits) - 1. But if the test with these values are sufficient, I'll modify the code.